kedro/framework/cli/project.py

```diff
@@ -199,6 +200,7 @@ def package(metadata: ProjectMetadata) -> None:
     help=PARAMS_ARG_HELP,
     callback=_split_params,
 )
+@click.option("--new_catalog", "new_catalog", is_flag=True, help=NEW_CATALOG_ARG_HELP)
```

Comment:
How does this flag play along with the `DATA_CATALOG_CLASS` defined in `settings.py`?

https://github.com/kedro-org/kedro/blob/7a16e1a48a2b1433a39b334787c534569abc4dbe/kedro/templates/project/%7B%7B%20cookiecutter.repo_name%20%7D%7D/src/%7B%7B%20cookiecutter.python_package%20%7D%7D/settings.py#L44-L46

Reply:
Is this suggestion just for the PoC or to be included in the final code? Adding a temporary flag just for the new catalog has a very low API surface to benefit ratio. Removing ANY flag from the CLI is a breaking change, including this. Could we just utilise the existing methodology as pointed by @astrojuanlu ?

Reply:
I did this temporarily to simplify running the project with both catalogs without modifying `DATA_CATALOG_CLASS` for the project before running it.  This flag and `DATA_CATALOG_CLASS_NEW` will go away later on.